oneview_redfish_toolkit/api/network_interface.py

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+
+# Copyright (2017) Hewlett Packard Enterprise Development LP
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+from oneview_redfish_toolkit.api.redfish_json_validator \
+    import RedfishJsonValidator
+
+
+class NetworkInterface(RedfishJsonValidator):
+    """Creates a NetworkInterface Redfish dict
+
+        Populates self.redfish with NetworkInterface data retrieved from
+        OneView
+    """
+
+    SCHEMA_NAME = 'NetworkInterface'
+
+    def __init__(self, device_id, server_hardware):
+        """NetworkInterface constructor
+
+            Populates self.redfish with the contents of server hardware dict
+            from Oneview
+
+            Args:
+                server_hardware: Oneview's server hardware dict
+        """
+        super().__init__(self.SCHEMA_NAME)
+        self.redfish["@odata.type"] = \
+            "#NetworkInterface.v1_1_0.NetworkInterface"
+        self.redfish["Id"] = device_id
+        index = int(device_id) - 1
```

Comment:
What if device_id is an invalid value? An **IndexError** will be raised. I'm considering this case in NetworkPortCollection.

Reply:
If you change the route this cast is no longer needed.

Reply:
Adding validation in the blueprint